src/routes/boot.ts

```diff
@@ -328,6 +336,49 @@ const getUser = (con: DataSource, userId: string): Promise<User> =>
     ],
   });
 
+const getMarketingCta = async (
+  con: DataSource,
+  log: FastifyBaseLogger,
+  userId: string,
+) => {
+  if (!userId) {
+    log.info('no userId provided when fetching marketing cta');
+    return null;
+  }
+
+  let marketingCta: MarketingCta | null = null;
+  const redisKey = generateStorageKey(
+    StorageTopic.Boot,
+    'marketing_cta',
+    userId,
+  );
+  const rawRedisValue = await getRedisObject(redisKey);
+
+  if (rawRedisValue === RedisMagicValues.SLEEPING) {
+    return null;
+  } else {
+    marketingCta = JSON.parse(rawRedisValue);
```

Comment:
Should we not simply return here already?

Reply:
No 😿
It's for when there is no redis key available, the redis response will return `null`, and in that case, we want to query the database and fill it